sycl/test-e2e/format.py

```diff
@@ -98,6 +98,47 @@ def getMatchedFromList(self, features, alist):
         except ValueError as e:
             raise ValueError("Error in UNSUPPORTED list:\n%s" % str(e))
 
+    def make_default_features_list(self, expr, triple, add_default=True):
+        ### EXCEPTIONS LIST
+        # TODO: Define elsewhere?
+        exceptions = {}
+        exceptions["spir64"]={
+                "cuda":False, "hip":False, "hip_amd":False, "hip_nvidia":False,
```

Comment:
not totally clear to me what this represents, maybe we could rename the var or add comments?

Reply:
> * The way tests are marked as unsupported is altered. Instead of testing what devices on the system are supported by the test, we test what triples the test can be built for. To do this we assume that  the triple supports all features in the `REQUIRES:` line, unless we know for certain it does not (i.e., for spir64 we know we are not supporting the hip backend). We also do this for `UNSUPPORTED:` and `XFAIL:`, but instead we assume the triple has no feature unless we know it always supports a given feature (i.e., nvptx64-nvidia-cuda will always use sg-32).

Basically this list is meant to represent the features for a triple (or the environment) that we always know the status of. So if we have a triple and can never have a given feature, we would add it to this dictionary as "false". That's what you see with "spir64" and cuda for example. We may have features that are always true for a triple, those would go in the dictionary as "True", like having "sg-32" for "nvptx64-nvidia-cuda". In the last case we may have features that may be true or false for a given triple, in this case we would not add it to the dictionary. In this case think for example "spir64" and the feature "level_zero", since we could be using the "OpenCL" backend instead.

This list just allows us to make the `REQUIRES:`, `UNSUPPORTED:` and `XFAIL:` statements be as permissive as possible while still taking into account combinations of features and triples that we know always/never happen.

Reply:
thanks for the explanation, that totally makes sense. maybe we could try to rework the code/add comments so that's more spelled out?

Reply:
added a comment about this list in the most recent commit